core/commands/storage/challenge/challenge.go

```diff
@@ -41,48 +41,86 @@ still store a piece of file (usually a shard) as agreed in storage contract.`,
 	}, StorageChallengeResponseCmd.Arguments...), // append pass-through arguments
 	RunTimeout: 20 * time.Second,
 	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		tm := timeEvaluate{}
+		tm.Init(6)
 		cfg, err := cmdenv.GetConfig(env)
 		if err != nil {
 			return err
 		}
 		if !cfg.Experimental.StorageClientEnabled {
 			return fmt.Errorf("storage client api not enabled")
 		}
+		tm.RecordTime("GetConfig")
 
 		n, err := cmdenv.GetNode(env)
 		if err != nil {
 			return err
 		}
 
+		tm.RecordTime("GetNode")
+
 		api, err := cmdenv.GetApi(env, req)
 		if err != nil {
 			return err
 		}
-
+		tm.RecordTime("GetApi")
 		// Check if peer is reachable
 		pi, err := remote.FindPeer(req.Context, n, req.Arguments[0])
 		if err != nil {
 			return err
 		}
+		tm.RecordTime("FindPeer")
 		// Pass arguments through to host response endpoint
 		resp, err := remote.P2PCallStrings(req.Context, n, api, pi.ID, "/storage/challenge/response",
 			req.Arguments[1:]...)
 		if err != nil {
 			return err
 		}
 
+		tm.RecordTime("P2PCall")
 		var scr StorageChallengeRes
 		err = json.Unmarshal(resp, &scr)
 		if err != nil {
 			return err
 		}
+		scr.timeEvaluate = append(scr.timeEvaluate, tm)
 		return cmds.EmitOnce(res, &scr)
 	},
 	Type: StorageChallengeRes{},
 }
 
+type timeEvaluate struct {
+	TmVal   []time.Time
+	TmIndex int
+	Event   []string
+}
+
+func (t *timeEvaluate) Init(size int) {
```

Comment:
I'd suggest using a dynamic size. In case somehow an additional RecordTime is called then you'd have to go back and change init size again..

Reply:
Yes, on that case, I change it to use append to use dynamic array, should be better.